### PR TITLE
Add custom themes support

### PR DESCRIPTION
**Pull request by sdex**
This PR adds a new option to the customization preferences - set light and dark themes. 

Create a new theme: 
1. Create a new file in the `res/values/` folder. The suggested filename format is `waterfox_theme_{theme_id}.xml`. 
2. Copy the existing theme, rename it, and set colors. 
3. Create a new color palette in the `net.waterfox.android.theme.WaterfoxThemeColors` file with the colors used in the XML file. 
4. Create the theme thumbnail in the `res/drawable/` folder. The suggested filename format is `theme_thumbnail_{theme_id}.xml`.
5. Add a new object to the `net.waterfox.android.theme.ThemeManager.THEMES` list. 

We have to duplicate the colors in the XML file (for views) and in the code (for Compose). 
To avoid duplicates it should be possible to read color attributes from XML and use them in composables (not recommended yet possible). 
Another option might be to create a script to generate both files from the set of colors. 

In this PR, I added 2 themes, they are for demonstration purposes only.

### CODE REVIEW

**Comment by sdex** [comment]
This is how the theme selection looks:

image

I added a few colors for example. 
Green and red have gradients, the rest are just plain colors. 

image

**Comment by MrAlex94** [comment]
Is it possible to have the changes rebased on top of the latest HEAD of `main`?

Regarding the first color icon, I was thinking a "transparent" or "neutral" should be the first color, that is, basically the default light/dark themes as they are now. Potentially it could be dynamically updated if the user has set dark or light mode below.

Everything else is great, I just need to collect the other `ids` that should probably change colour as well.

**Comment by sdex** [comment]
I had to use merge. 

I added the first color (it's default theme color with the name `layer1`). I tried to set the transparent, but it looked odd.

File: app/src/main/java/net/waterfox/android/theme/ThemeManager.kt [hunk 1]
```diff
@@ -135,3 +139,46 @@ class DefaultThemeManager(
...
+    return COLOR_SCHEMES.find { it.id == colorScheme }!!
+}
+
+val COLOR_SCHEMES = listOf(
+    WaterfoxThemeColorScheme(
+        id = -1,
+        lightColors = lightColorPalette,
+        darkColors = darkColorPalette,
+        resourceId = R.style.NormalTheme,
+        primaryColor = lightColorPalette.layer1,
```
Comment (MrAlex94) [inline app/src/main/java/net/waterfox/android/theme/ThemeManager.kt:154]: Is it possible for this to change depending on whether dark or light mode is selected? Currently it stays white from my brief testing.

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1770,4 +1776,11 @@
     <string name="privacy_content_close_button_content_description">Close</string>
 
     <string name="waterfox_suggest_header">Waterfox Suggest</string>
+
+    <!-- Waterfox themes -->
```
Comment (MrAlex94) [inline app/src/main/res/values/strings.xml:1780]: Are these unused?

**Comment by MrAlex94** [comment]
Thanks, this is looking great now.

The last remaining parts:

* Is it possible to have the "purple" items below match the selected theme, when the default isn't selected?
Screenshot_20240208_115740
* Is it possible for the preview colours to match the gradients? For example they're quite flat at the moment but would expect something similar to:
image

**Comment by sdex** [comment]
@MrAlex94 I updated the theme selection screen and themes to match the selected theme color. 

Screenshot_20240208_180035_Waterfox

**Comment by MrAlex94** [comment]
That's great, thanks!

**Comment by MrAlex94** [comment]
https://github.com/WaterfoxCo/Waterfox-Android/commit/e58ffc5177690cb6dce012da797a0e7b374731d1

**Pull request closed without merging**